Source/WebKit/UIProcess/WebProcessPool.h

```diff
@@ -344,7 +345,7 @@ class WebProcessPool final
 
     void reportWebContentCPUTime(Seconds cpuTime, uint64_t activityState);
 
-    Ref<WebProcessProxy> processForSite(WebsiteDataStore&, const std::optional<WebCore::Site>&, WebProcessProxy::LockdownMode, const API::PageConfiguration&); // Will return an existing one if limit is met or due to caching.
+    Ref<WebProcessProxy> processForSite(WebsiteDataStore&, const std::optional<WebCore::Site>&, WebProcessProxy::LockdownMode, const API::PageConfiguration&, WebCore::ProcessSwapDisposition = WebCore::ProcessSwapDisposition::None, const URL& destinationURL = URL()); // Will return an existing one if limit is met or due to caching.
```

Comment:
It is confusing to pass in a Site and a URL that should be the same.  We should pass one or the other but not both.  Maybe rename the function too.  processForNavigationToURL?